deepfence_server/pkg/integration/teams/teams.go

```diff
@@ -35,35 +38,51 @@ func (t Teams) FormatMessage(message map[string]interface{}, position int) strin
 }
 
 func (t Teams) SendNotification(ctx context.Context, message string, extras map[string]interface{}) error {
+	t.client = utils.GetHttpClient()
+
 	var msg []map[string]interface{}
 	d := json.NewDecoder(strings.NewReader(message))
 	d.UseNumber()
 	if err := d.Decode(&msg); err != nil {
+		log.Info().Msgf("Failed to unmarshal message for teams: %v", err)
 		return err
 	}
 
-	startIndex := 0
-	endIndex := BatchSize
-	if endIndex > len(msg) {
-		endIndex = len(msg)
+	numRoutines := (len(msg) / 10)
+	if numRoutines == 0 {
+		numRoutines = 1
+	} else if numRoutines > 10 {
+		numRoutines = 10
 	}
-	if err := t.sendNotification(msg[startIndex:endIndex]); err != nil {
-		return err
+
+	senderChan := make(chan *Payload, 500)
+	var wg sync.WaitGroup
+	for i := 0; i < numRoutines; i++ {
+		wg.Add(1)
+		go t.Sender(senderChan, &wg)
 	}
+
+	defer func() {
+		close(senderChan)
+		wg.Wait()
+		log.Info().Msgf("Teams::SendNotification complete, numRoutines:%d", numRoutines)
+	}()
+
+	startIndex := 0
+	endIndex := BatchSize
```

Comment:
Are we guaranteed that `len(msg)` is always a multiple of `BatchSize`?
If not, this can panic

Reply:
Hmmm... good catch. 
Fixed this